CAIPs/caip-x.md

```diff
@@ -0,0 +1,99 @@
+
+### Motivation
+
+First specifically why switch from an injected provider to `externally_connectable`
+
+#### Enhanced Security Benefits
+
+ **Reduced Permissions Requirement**: Utilizing the `externally_connectable` API allows wallet extensions communicate with websites, without requiring broad permissions to read and modify all data on the websites a user visits as is currently the case with the typical injected provider used by most extension Wallets. For instance the `externally_connectable` method does not allow modification of web page content. This reduces the level of trust and permission users must grant to the extension, minimizing potential security risks if the extension or its updates are compromised.
+    
+**Reduced Fingerprinting Risk**: The traditional method of injecting a JavaScript provider into every webpage a user visits exposes the user to increased fingerprinting risks. Each script injection can potentially be detected by malicious websites aiming to track users by identifying unique behaviors in their browser environments.
+
+#### Enhanced Performance Benefits
+
+**Fewer Blocking Scripts**: Injected scripts often modify the DOM or execute operations that can block or slow down page loading, especially if they are not efficiently optimized. By using the `externally_connectable` approach, extensions do not inject any JavaScript into the pages, thereby avoiding the introduction of blocking scripts that can degrade the performance of web pages.
+    
+ **Optimized Resource Usage**: The `externally_connectable` method focuses on direct communication through message passing, which is generally more resource-efficient than script injection. This means less CPU and memory usage as the communication is handled by the browser’s built-in mechanisms rather than through additional JavaScript execution on the page.
+
+**Faster Page Load Times**: Without the overhead of injecting and running additional scripts on every page, pages load faster, providing a smoother and more responsive user experience. This performance improvement is critical for user retention and satisfaction, especially on resource-intensive applications or slower devices.
+
+#### So Why Wallet Discovery?
+
+Given the benefits of `externally_connectable` we need to solve for a key area where it fails to match the injected provider pattern.  
+`externally_connectable` requires that the webpage attempting to connect with the wallet know the extension's canonical extension id. Currently, there is no standard method for these wallets to provide their extension IDs to dApps. A standardized approach will improve interoperability, security, and user experience by ensuring that dApps can reliably identify and communicate with the intended wallet.
```

Comment:
Thoughts on mentioning why this is preferable to looking up extension Ids from a centralized repo?

Reply:
added!